### PR TITLE
Test: UTF-8 vs latin-1 regression

### PR DESCRIPTION
**Pull request by joedevivo**
We need to determine wether or not Riak 1.4's app.config and vm.args file could accept UTF-8 values or if they are restricted to latin-1. For example, multi backend bucket names.

If riak 1.4 can accept UTF-8 values, cuttlefish needs to be able to as well. If they can't , then it's desirable for cuttlefish to be able to detect non latin1 files and print an error message, but that might be a 2.0.1 fix.

### CODE REVIEW

**Comment by joedevivo** [comment]
In riak.conf 

``` properties
##
## Default: ./data
##
## Acceptable values:
##   - the path to a directory
platform_data_dir = ./dataŒ
```

``` erlang
{platform_data_dir,[46,47,100,97,116,97,338]},
```

```
drwxr-xr-x   4 joe  staff   136 Apr 22 08:44 dataŒ
```

```
2014-04-22 08:44:48.167 [warning] <0.216.0>@riak_core_ring_manager:reload_ring:355 No ring file available.
2014-04-22 08:44:48.261 [error] <0.222.0> CRASH REPORT Process <0.222.0> with 0 neighbours exited with reason: bad argument in call to eleveldb:destroy([46,47,100,97,116,97,338,47,99,108,117,115,116,101,114,95,109,101,116,97,47,116,114,101,101,115], []) in hashtree:destroy/1 line 262 in gen_server:init_it/6 line 328
2014-04-22 08:44:48.261 [error] <0.202.0> Supervisor riak_core_sup had child riak_core_metadata_hashtree started with riak_core_metadata_hashtree:start_link() at undefined exit with reason bad argument in call to eleveldb:destroy([46,47,100,97,116,97,338,47,99,108,117,115,116,101,114,95,109,101,116,97,47,116,114,101,101,115], []) in hashtree:destroy/1 line 262 in context start_error
2014-04-22 08:44:48.264 [error] <0.200.0> CRASH REPORT Process <0.200.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,riak_core_metadata_hashtree,{badarg,[{eleveldb,destroy,[[46,47,100,97,116,97,338,47,99,108,117,115,116,101,114,95,109,101,116,97,47,116,114,101,101,115],[]],[]},{hashtree,destroy,1,[{file,"src/hashtree.erl"},{line,262}]},{hashtree_tree,create_node,2,[{file,"src/hashtree_tree.erl"},{line,457}]},{hashtree_tree,new,2,[{file,"src/hashtree_tree.erl"},{line,187}]},{riak_core_metadata_hashtree,init,1,[{file,"src/riak_core_metadata_hashtree.erl"},{line,169}]},{gen_server,...},...]}}},...} in application_master:init/4 line 133
2014-04-22 08:44:48.265 [info] <0.7.0> Application riak_core exited with reason: {{shutdown,{failed_to_start_child,riak_core_metadata_hashtree,{badarg,[{eleveldb,destroy,[[46,47,100,97,116,97,338,47,99,108,117,115,116,101,114,95,109,101,116,97,47,116,114,101,101,115],[]],[]},{hashtree,destroy,1,[{file,"src/hashtree.erl"},{line,262}]},{hashtree_tree,create_node,2,[{file,"src/hashtree_tree.erl"},{line,457}]},{hashtree_tree,new,2,[{file,"src/hashtree_tree.erl"},{line,187}]},{riak_core_metadata_hashtree,init,1,[{file,"src/riak_core_metadata_hashtree.erl"},{line,169}]},{gen_server,...},...]}}},...}
```

[46,47,100,97,116,97,338,47,99,108,117,115,116,101,114,95,109,101,116,97,47,116,114,101,101,115] = "./dataŒ/cluster_meta/trees"

**Comment by joedevivo** [comment]
``` properties
distributed_cookie = riakŒ
```

```
09:01:41.482 [info] /Users/joe/Downloads/riak-ee-2.0.0beta1/bin/../etc/advanced.config detected, overlaying proplists
escript: exception error: bad argument
  in function  io_lib:format/2
     called as io_lib:format("~s ~s",['-setcookie',[114,105,97,107,338]])
  in call from cuttlefish_vmargs:stringify_line/2 (src/cuttlefish_vmargs.erl, line 17)
  in call from cuttlefish_vmargs:'-stringify/1-lc$^0/1-0-'/1 (src/cuttlefish_vmargs.erl, line 13)
  in call from cuttlefish_vmargs:'-stringify/1-lc$^0/1-0-'/1 (src/cuttlefish_vmargs.erl, line 13)
  in call from cuttlefish_escript:engage_cuttlefish/1 (src/cuttlefish_escript.erl, line 359)
  in call from cuttlefish_escript:generate/1 (src/cuttlefish_escript.erl, line 235) 
```

**Comment by joedevivo** [comment]
so, as far as data dir goes, 1.4 creates the wrong name, but still starts. 2.0 creates the right name, but riak can't start.

**Comment by joedevivo** [comment]
```
➜  riak git:(develop) ✗ ./bin/riak console
13:14:28.422 [error] /Users/joe/dev/basho/riak_ee/rel/riak/bin/../etc/riak.conf: Error converting value on line #200 to latin1
Error generating config with cuttlefish
  run `riak config generate -l debug` for more information.
```

**Comment by joedevivo** [comment]
Eunit's failing on the builder, but passes locally. will investigate.

**Comment by joedevivo** [comment]
All good now. didn't check in the test fixtures

File: src/conf_parse.peg [hunk 1]
```diff
@@ -48,7 +48,15 @@ key <- head:word tail:("." word)* %{
 %};
 
 %% A value is any character, with trailing whitespace stripped.
-value <- (!((ws* crlf) / comment) .)+ `unicode:characters_to_list(iolist_to_binary(Node))`;
+value <- (!((ws* crlf) / comment) .)+ %{
+  case
```
Comment (seancribbs) [inline src/conf_parse.peg]: What's with the weird line-wrapping and indentation here?

File: src/cuttlefish_schema.erl [hunk 2]
```diff
@@ -122,7 +122,7 @@ count_mappings(Mappings) ->
 file(Filename, Schema) ->
     {ok, B} = file:read_file(Filename),
     %% TODO: Hardcoded utf8
```
Comment (seancribbs) [inline src/cuttlefish_schema.erl]: Could remove the TODO here.

Comment (seancribbs) [inline src/cuttlefish_schema.erl]: Maybe replace it with a comment about why latin1 is necessary?

**Comment by seancribbs** [comment]
:+1: 8ef24f7

**Comment by joedevivo** [comment]
@borshop merge

**Pull request merged**